File: 3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function (s) {
    if (s.length < 3) {
        return s.length
    }
    let count = new Map()
    let ans = 0
    for (let i = 0; i < s.length; i++) {
        count.set(s[i], (count.get(s[i]) || 0) + 1)
    }
    for (let [key, val] of count) {
        while (val >= 3) {
            ans += 2
            val -= 2
        }
    }
    ans = s.length - ans
    return ans
};

File: 3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.py
class Solution:
    def minimumLength(self, s: str) -> int:
        if len(s) < 3:
            return len(s)
        
        count = {}
        ans = 0
        
        for char in s:
            count[char] = count.get(char, 0) + 1
        
        for val in count.values():
            while val >= 3:
                ans += 2
                val -= 2
        
        ans = len(s) - ans
        return ans

